store/postgres/src/store.rs

```diff
@@ -181,92 +262,225 @@ impl Store {
             .execute(&*self.conn.lock().unwrap())
             .unwrap();
     }
-}
-
-impl BasicStore for Store {
-    fn get(&self, key: StoreKey) -> Result<Entity, ()> {
-        debug!(self.logger, "get"; "key" => format!("{:?}", key));
 
+    /// Gets an entity from Postgres, returns an entity with just an ID if none is found.
+    fn get_entity(
+        &self,
+        conn: &PgConnection,
+        op_subgraph: &String,
+        op_entity: &String,
+        op_id: &String,
+    ) -> Result<Entity, Error> {
         use db_schema::entities::dsl::*;
 
-        // Use primary key fields to get the entity; deserialize the result JSON
-        entities
-            .find((key.id, key.subgraph, key.entity))
+        match entities
+            .find((op_id, op_subgraph, op_entity))
             .select(data)
-            .first::<serde_json::Value>(&*self.conn.lock().unwrap())
-            .map(|value| {
-                serde_json::from_value::<Entity>(value).expect("Failed to deserialize entity")
-            }).map_err(|_| ())
+            .first::<serde_json::Value>(conn)
+        {
+            Ok(json) => serde_json::from_value::<Entity>(json).map_err(|e| {
+                format_err!(
+                    "Encountered invalid entity ({}, {}, {}) in the store: {}",
+                    op_subgraph,
+                    op_entity,
+                    op_id,
+                    e
+                )
+            }),
+            Err(_) => Ok(Entity::from(vec![("id", Value::from(op_id.clone()))])),
+        }
     }
 
-    fn set(
-        &mut self,
-        key: StoreKey,
-        input_entity: Entity,
-        input_event_source: EventSource,
-    ) -> Result<(), ()> {
-        debug!(self.logger, "set"; "key" => format!("{:?}", key));
-
+    /// Applies a set operation in Postgres.
+    fn apply_set_operation(
+        &self,
+        conn: &PgConnection,
+        operation: EntityOperation,
+        block_ptr_to: &EthereumBlockPointer,
```

Comment:
`EthereumBlockPointer` is Copy so no ref needed here or in other function params